.github/workflows/tsbench-coverage.yaml

```diff
@@ -0,0 +1,204 @@
+          wget -q -O postgresql.tar.bz2 \
+          https://ftp.postgresql.org/pub/source/v${{ steps.config.outputs.PG15_LATEST }}/postgresql-${{ steps.config.outputs.PG15_LATEST }}.tar.bz2
+          mkdir -p ~/$PG_SRC_DIR
+          tar --extract --file postgresql.tar.bz2 --directory ~/$PG_SRC_DIR --strip-components 1
+          cd ~/$PG_SRC_DIR
+          ./configure --prefix=$HOME/$PG_INSTALL_DIR --with-openssl \
+            --without-readline --without-zlib --without-libxml \
+            --enable-cassert --enable-debug
+          make -j$(nproc)
+
+      - name: Save PostgreSQL Cache
+        if: steps.restore-postgresql-cache.outputs.cache-matched-key == ''
+        id: save-postgresql-cache
+        uses: actions/cache/save@v3
+        with:
+          path: ~/${{ env.PG_SRC_DIR }}
+          key: "${{ steps.cache-key.outputs.key }}"
+
+      - name: Install PostgreSQL
+        run: |
+          make -C ~/$PG_SRC_DIR install
+
+      - name: Upload config.log
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: config.log for PostgreSQL
+          path: ~/${{ env.PG_SRC_DIR }}/config.log
+
+      - name: Build TimescaleDB
+        run: |
+          set -e
+
+          cmake -B build -S . -DASSERTIONS=ON -DCODECOVERAGE=1 -DLINTER=OFF \
+            -DCMAKE_VERBOSE_MAKEFILE=1 \
+            -DWARNINGS_AS_ERRORS=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+            -DPG_PATH=$HOME/$PG_INSTALL_DIR
+
+          make -C build -j$(nproc) install
+
+      - name: initdb
+        run: |
+          set -xeu
+
+          export PGDATA=db
+          export PGPORT=5432
+          export PGDATABASE=postgres
+          export PATH=$HOME/$PG_INSTALL_DIR/bin:$PATH
+          initdb
+
+          go install github.com/timescale/timescaledb-tune/cmd/timescaledb-tune@latest
+
+          ~/go/bin/timescaledb-tune --yes
+
+          echo "statement_timeout=90s" >> $PGDATA/postgresql.conf
+
+      - name: Checkout tsbench
+        uses: actions/checkout@v3
+        with:
+          repository: 'timescale/tsbench'
+          path: 'tsbench'
+          token: ${{ secrets.ORG_AUTOMATION_TOKEN }}
+
+      - name: Prepare tsbench
+        run: |
+          set -xeu
+
+          cd tsbench
+          python3 -m venv .venv
+          source .venv/bin/activate
+          pip install -e .
+
+      - name: Run tsbench
+        run: |
+          set -xeu
+
+          export PGDATA=db
+          export PGPORT=5432
+          export PGDATABASE=postgres
+          export PATH=$HOME/$PG_INSTALL_DIR/bin:$PATH
+          pg_ctl -l postmaster.log start
+
+          psql -c "create extension timescaledb;"
+
+          cd tsbench
+          source .venv/bin/activate
+          tsbench-deployer --verbose --cache-s3-datasets
+          LANG=C LC_NUMERIC=C tsbench --max-executions=1 --verbose --verbose \
+            --with-connection pgsq://localhost:5432/postgres
+
+          # Postgres should survive.
+          psql -c "select 1;"
```

Comment:
What is the reason for this test? Just to make sure PostgreSQL did not crash during the benchmark execution?

Reply:
Yes.